services/121-service/src/program-financial-service-provider-configurations/mappers/program-financial-service-provider-configuration.mapper.ts

```diff
@@ -14,16 +19,100 @@ export class ProgramFinancialServiceProviderConfigurationMapper {
   public static mapEntitytoDto(
     entity: ProgramFinancialServiceProviderConfigurationEntity,
   ): ProgramFinancialServiceProviderConfigurationResponseDto {
-    const dto = new ProgramFinancialServiceProviderConfigurationResponseDto();
-
-    dto.programId = entity.programId;
-    dto.financialServiceProviderName = entity.financialServiceProviderName;
-    dto.name = entity.name;
-    dto.label = entity.label;
-    const financialServiceProvider = FINANCIAL_SERVICE_PROVIDERS.find(
-      (fsp) => fsp.name === entity.financialServiceProviderName,
+    const financialServiceProvider = findFinancialServiceProviderByNameOrFail(
+      entity.financialServiceProviderName,
     );
-    dto.financialServiceProvider = financialServiceProvider;
+    const dto: ProgramFinancialServiceProviderConfigurationResponseDto = {
+      programId: entity.programId,
+      financialServiceProviderName: entity.financialServiceProviderName,
+      name: entity.name,
+      label: entity.label,
+      financialServiceProvider,
+      properties:
+        ProgramFinancialServiceProviderConfigurationMapper.mapPropertyEntitiesToDtos(
+          entity.properties,
+        ),
+    };
     return dto;
   }
+
+  public static mapDtoToEntity(
+    dto: CreateProgramFinancialServiceProviderConfigurationDto,
+    programId: number,
+  ): ProgramFinancialServiceProviderConfigurationEntity {
+    const entity = new ProgramFinancialServiceProviderConfigurationEntity();
+    entity.programId = programId;
+    entity.financialServiceProviderName = dto.financialServiceProviderName;
+    entity.name = dto.name;
+    entity.label = dto.label;
+    return entity;
+  }
+
+  public static mapPropertyEntitiesToDtos(
+    properties?: ProgramFinancialServiceProviderConfigurationPropertyEntity[],
+  ): ProgramFinancialServiceProviderConfigurationResponsePropertyDto[] {
+    if (!properties) {
+      return [];
+    }
+    return properties.map((property) =>
+      ProgramFinancialServiceProviderConfigurationMapper.mapPropertyEntityToDto(
+        property,
+      ),
+    );
+  }
+
+  public static mapPropertyEntityToDto(
+    property: ProgramFinancialServiceProviderConfigurationPropertyEntity,
+  ): ProgramFinancialServiceProviderConfigurationResponsePropertyDto {
+    return {
+      name: property.name,
+      updated: property.updated,
+    };
+  }
+
+  public static mapPropertyDtosToEntities(
+    dtos: CreateProgramFinancialServiceProviderConfigurationPropertyDto[],
+    programFinancialServiceProviderConfigurationId: number,
+  ): ProgramFinancialServiceProviderConfigurationPropertyEntity[] {
+    return dtos.map((dto) =>
+      ProgramFinancialServiceProviderConfigurationMapper.mapPropertyDtoToEntity(
+        dto,
+        programFinancialServiceProviderConfigurationId,
+      ),
+    );
+  }
+
+  public static mapPropertyDtoToEntity(
```

Comment:
is public only for the purpose of the unit test, which shouldn't be the case. 

Also, how did you choose overall which public methods of this file to unit test, and which not?

Reply:
I thought mostly about which unit tests would have too much overlap and decided based on that